cmd.go

```diff
@@ -0,0 +1,79 @@
+package gps
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"time"
+)
+
+// monitoredCmd wraps a cmd and will keep monitoring the process until it
+// finishes or a certain amount of time has passed and the command showed
+// no signs of activity.
+type monitoredCmd struct {
+	cmd     *exec.Cmd
+	timeout time.Duration
+	buf     *activityBuffer
+}
+
+func newMonitoredCmd(cmd *exec.Cmd, timeout time.Duration) *monitoredCmd {
+	buf := newActivityBuffer()
+	cmd.Stderr = buf
+	cmd.Stdout = buf
+	return &monitoredCmd{cmd, timeout, buf}
+}
+
+// run will wait for the command to finish and return the error, if any. If the
+// command does not show any activity for more than the specified timeout the
+// process will be killed.
+func (c *monitoredCmd) run() error {
+	ticker := time.NewTicker(c.timeout)
+	done := make(chan error, 1)
+	defer ticker.Stop()
+	go func() { done <- c.cmd.Run() }()
+
+	for {
+		select {
+		case <-ticker.C:
+			if c.hasTimedOut() {
+				if err := c.cmd.Process.Kill(); err != nil {
+					return fmt.Errorf("error killing process after command timed out: %s", err)
```

Comment:
Let's create a special (unexported) type to represent this error. It doesn't need to be fancy, just serve the immediate needs here. #20 is intended to be a spot where we do a wholistic revisiting of the errors in gps, at which time we'll give it a more consistent treatment.